### PR TITLE
Fix elect command bug

### PR DESCRIPTION
**Pull request by codings-dan**
### What changes are proposed in this pull request?

Fix elect command bug

### Why are the changes needed?
If we start on the same machine, for example ,localhost, the judgment will be wrong, and we need to continue to judge whether the port are consistent.

### Does this PR introduce any user facing changes?
No

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java [hunk 1]
```diff
@@ -81,7 +81,8 @@ public int run(CommandLine cl) throws IOException {
         } catch (UnavailableException e) {
           return false;
         }
-        return leaderAddress.getHostName().equals(address.getHost());
+        return leaderAddress.getHostName().equals(address.getHost())
+                && leaderAddress.getPort() == address.getRpcPort();
```
Comment (maobaolong) [inline shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java:85]: getRpcPort return `19998` leaderAddress.getPort() is `19200`, so it never be equal

**Review by maobaolong — APPROVED** [review]
It LGTM, thank you for your improvement.

File: shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java [hunk 2]
```diff
@@ -68,20 +66,24 @@ public int run(CommandLine cl) throws IOException {
...
-          leaderAddress = inquireClient.getPrimaryRpcAddress();
+          GetQuorumInfoPResponse quorumInfo = jmClient.getQuorumInfo();
+
+          Optional<QuorumServerInfo>
+                  leadingMasterInfoOpt = quorumInfo.getServerInfoList().stream()
+                  .filter(QuorumServerInfo::getIsLeader).findFirst();
+          String leadingMasterAddr = leadingMasterInfoOpt.isPresent()
+                  ? netAddressToString(leadingMasterInfoOpt.get().getServerAddress()) : "UNKNOWN";
+          return leadingMasterAddr.equals(netAddressToString(address));
         } catch (UnavailableException e) {
```
Comment (ggezer) [inline shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java:82]: no need for this while catching `AlluxioStatusException`.

File: shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java [hunk 3]
```diff
@@ -68,20 +66,24 @@ public int run(CommandLine cl) throws IOException {
...
         try {
-          leaderAddress = inquireClient.getPrimaryRpcAddress();
+          GetQuorumInfoPResponse quorumInfo = jmClient.getQuorumInfo();
+
+          Optional<QuorumServerInfo>
+                  leadingMasterInfoOpt = quorumInfo.getServerInfoList().stream()
+                  .filter(QuorumServerInfo::getIsLeader).findFirst();
+          String leadingMasterAddr = leadingMasterInfoOpt.isPresent()
+                  ? netAddressToString(leadingMasterInfoOpt.get().getServerAddress()) : "UNKNOWN";
+          return leadingMasterAddr.equals(netAddressToString(address));
```
Comment (ggezer) [inline shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java:81]: why convert to `String`?
`NetAddress`es should be comparable.

**Review by ggezer — CHANGES_REQUESTED** [review]
Thanks for the fix. Left few minor requests.

File: shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java (hunk 3 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java:81]: Sure, @codings-dan you can use `NetAddress#equals` to compare these two address are the same.

**Comment by codings-dan** [comment]
@ggezer PTAL，thx

File: shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java [hunk 4]
```diff
@@ -76,11 +74,9 @@ public int run(CommandLine cl) throws IOException {
...
                   leadingMasterInfoOpt = quorumInfo.getServerInfoList().stream()
                   .filter(QuorumServerInfo::getIsLeader).findFirst();
-          String leadingMasterAddr = leadingMasterInfoOpt.isPresent()
-                  ? netAddressToString(leadingMasterInfoOpt.get().getServerAddress()) : "UNKNOWN";
-          return leadingMasterAddr.equals(netAddressToString(address));
-        } catch (UnavailableException e) {
-          return false;
+          NetAddress leaderAddress = leadingMasterInfoOpt.isPresent()
+                  ? leadingMasterInfoOpt.get().getServerAddress() : null;
+          return leaderAddress.equals(address);
```
Comment (maobaolong) [inline shell/src/main/java/alluxio/cli/fsadmin/journal/QuorumElectCommand.java:79]:
```suggestion
          return address.equals(leaderAddress);
```

**Review by ggezer — APPROVED** [review]
Thanks for the fix.

**Comment by apc999** [comment]
@jenoudet can you take a look? thanks

**Review by jenoudet — APPROVED** [review]
Thanks for the fix.

**Comment by ggezer** [comment]
alluxio-bot, merge this please

**Pull request merged**